apollo/handlers/user.py

```diff
@@ -59,27 +58,19 @@ def patch_user(user_data: UpdateUserSchema, request: Request,
 
 def update_user_password_and_data(user, data):
     data = copy.copy(data)
-    error = None
     if not compare_plaintext_to_hash(data['old_password'],
                                      user.password_hash,
                                      user.password_salt):
-        error = JSONResponse(
-            status_code=400,
-            content={
-                'old_password': {
-                    'msg': 'Invalid password',
-                    'type': 'value_error'
-                }
-            }
-        )
+        raise RequestValidationErrorWrapper(
+            "Invalid password", ('old_password'))
 
     data['password_hash'], data['password_salt'] = hash_plaintext(
         data.pop('password'))
 
-    return data, error
+    return data
 
 
-@router.delete('/user/{user_id}', status_code=204, permission='user.delete')
+@ router.delete('/user/{user_id}', status_code=204, permission='user.delete')
```

Comment:
```suggestion
@router.delete('/user/{user_id}', status_code=204, permission='user.delete')
```
etc for the other changes.